src/main/scala/mesosphere/marathon/api/akkahttp/v2/EventsController.scala

```diff
@@ -52,15 +52,19 @@ class EventsController(
     authenticated.apply { implicit identity =>
       authorized(ViewResource, AuthorizedResource.Events).apply {
         parameters('event_type.*) { events =>
-          extractClientIP { clientIp =>
-            complete {
-              Source
-                .fromGraph[MarathonEvent, NotUsed](
-                  new EventStreamSourceGraph(eventBus, conf.eventStreamMaxOutstandingMessages(), clientIp)
-                )
-                .filter(isAllowed(events.toSet))
-                .map(event => ServerSentEvent(`type` = event.eventType, data = Json.stringify(Formats.eventToJson(event))))
-                .keepAlive(5.second, () => ServerSentEvent.heartbeat)
+          parameter("plan-format".?) { light =>
+            extractClientIP { clientIp =>
+              complete {
+                Source
```

Comment:
Do you think it make sense to move this to a different method?

Reply:
I am indifferent on this. Yes, the logic is a bit mixed, but on the other hand with the trick you suggested above the code becomes quite small and easy to understand :man_shrugging: